hare/broker.go

```diff
@@ -308,9 +309,12 @@ func (b *Broker) Register(id instanceID) (chan *Msg, error) {
 		b.updateLatestLayer(id)
 
 		if len(b.outbox) >= b.limit {
-			resErr <- errTooMany
-			resCh <- nil
-			return
+			//unregister the earliest layer to make space for the new layer
+			//cannot call unregister here because unregister blocks and this would cause a deadlock
+			id := b.minDeleted + 1
```

Comment:
Don't overload the `id` variable here, it makes this code hard to read. Use a different name.